launchpod/src/test/java/com/google/launchpod/FormHandlerServletTest.java

```diff
@@ -0,0 +1,403 @@
+    */
+    private String createUrl(String title, String id) {
+        String url = "<div><p>" + title + "URL: <a href=\"https://launchpod-step18-2020.appspot.com/?id=" + id + "\">" + id + "</a></p></div>";
+        return url;
+    }
+
+    /**
+     * Create RSS XML string given a user feed entity.
+     * @return xml String
+     * @throws IOException
+    */
+    private static String xmlString(RSS rssFeed) throws IOException {
+        XmlMapper xmlMapper = new XmlMapper();
+        String xmlString = xmlMapper.writeValueAsString(rssFeed);
+        return xmlString;
+    }
+
+    /**
+     * Asserts that doPost() takes in form inputs from client and successfully stores that information in a Datastore entity.
+     * @throws IOException
+    */
+    @Test
+    public void doPost_StoresCorrectFormInput() throws IOException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        when(request.getParameter(PODCAST_TITLE)).thenReturn(TEST_PODCAST_TITLE);
+        when(request.getParameter(MP3_LINK)).thenReturn(TEST_MP3_LINK);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doPost(request, response);
+    
+        assertEquals(1, ds.prepare(new Query(USER_FEED)).countEntities(withLimit(10)));
+
+        Query query = new Query("UserFeed");
+        Entity entity = ds.prepare(query).asSingleEntity();
+
+        assertEquals(entity.getProperty(PODCAST_TITLE).toString(), TEST_PODCAST_TITLE);
+        assertEquals(entity.getProperty(MP3_LINK).toString(), TEST_MP3_LINK);
+    }
+
+    /**
+     * Asserts that doPost() takes in form inputs from client, successfully stores that information in a Datastore entity, 
+     * and returns a URL link to the generated RSS feed.
+     * @throws IOException
+    */
+    @Test
+    public void doPost_ReturnsCorrectUrl() throws IOException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        when(request.getParameter(PODCAST_TITLE)).thenReturn(TEST_PODCAST_TITLE);
+        when(request.getParameter(MP3_LINK)).thenReturn(TEST_MP3_LINK);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doPost(request, response);
+    
+        assertEquals(1, ds.prepare(new Query(USER_FEED)).countEntities(withLimit(10)));
+
+        Query query = new Query("UserFeed");
+        Entity entity = ds.prepare(query).asSingleEntity();
+
+        assertEquals(entity.getProperty(PODCAST_TITLE).toString(), TEST_PODCAST_TITLE);
+        assertEquals(entity.getProperty(MP3_LINK).toString(), TEST_MP3_LINK);
+        RSS rssFeed = new RSS(TEST_PODCAST_TITLE, TEST_MP3_LINK);
+        TEST_XML_STRING = xmlString(rssFeed);
```

Comment:
TEST_XML_STRING is a static final - it shouldn't be possible to reassign it here. does this code run?

Reply:
Making the variable local to each test!